docs/spark-qualification-tool.md

```diff
@@ -482,35 +516,98 @@ The HTML view renders the detailed information into tables that allow following
+   The table has cascaded _searchPanes_, which means that the table allows the panes
+   to be filtered based on the values selected in the other panes.  
+   There are three searchPanes:
+    1. "_Is Stage Estimated_": it splits the stages into two groups based on whether the stage duration time was estimated
+       or not.
+    2. "_Speed-up_": groups the stages by their "average speed-up factor". Each stage can belong to one of the following
+       predefined speed-up ranges: `1.0 (No Speed-up)`; `]1.0, 1.3[`; `[1.3, 2.5[`; `[2.5, 5[`; and `[5, _]`. The
+       search-pane does not show a range bucket if its count is 0.
+    3. "_Tasks GPU Support_": this filter can be used to find stages having all their execs supported by the GPU.
+6. **Execs Details Table**: lists all the app stages with set of columns listed in [Execs report](#execs-report)
+   section. The HTML5 export button saves the table to CSV file into the browser's default
    download folder.
-
-![Qualification-HTML-Recommendation-View](img/Tools/qualification-tool-recommendation-header-01.png)
+   ![Qualification-HTML-App-Details-View-Execs](img/Tools/qualification-tool-app-view-03.png)
+   The table has cascaded _searchPanes_, which means that the table allows the panes
+   to be filtered based on the values selected in the other panes.  
+   There are three _searchPanes_:
+    1. "_Exec_": filters the rows by exec name. This filter also allows text searching by typing into the filter-title as
+       a text input.
+    2. "_Speed-up_": groups the stages by their "average speed-up factor". Each stage can belong to one of the following
+       predefined speed-up ranges: `1.0 (No Speed-up)`; `]1.0, 1.3[`; `[1.3, 2.5[`; `[2.5, 5[`; and `[5, _]`. The
+       search-pane does not show a range bucket if its count is 0.
+    3. "_GPU Support_": filters the execs whether an exec is supported by GPU or not.
+    4. "_Stage ID_": filters rows by the stage ID. It also allows text-searching by typing into the filter-title as a text
+       input.
+    5. "_Is Exec Removed_": filters rows that were removed from the migrated plan.
+   
 
 #### Raw Data
 
-`raw.html` displays the same all the fields listed in "_Detailed App Report_" in more readable format.
-Columns representing "_time duration_" are rounded to nearest "ms", "seconds", "minutes", and "hours".  
+`raw.html` displays all the fields listed in "_Detailed App Report_" in more readable format.
+Columns representing "_time duration_" are rounded to nearest "ms", "seconds", "minutes", and "hours".
 The search box will match on multiple columns including: "_App ID_", "_App Name_", "_Recommendation_",
 "_User Name_", "_Unsupported Write Data Format_", "_Complex Types_", "_Nested Complex Types_", and "_Read Schema_".
 The detailed table can also be exported as `Qualification Tool Dashboard – Raw Data.csv`.
 
+Note that this table has too many columns. Therefore, the UI application dynamically optimizes the layout of the table
```

Comment:
Instead of "too many columns", reword to say "more columns than can fit in a normal browser window."